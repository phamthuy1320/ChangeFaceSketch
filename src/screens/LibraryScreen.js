import React,{useState, useEffect} from 'react';
import {
    View, 
    Text, 
    StyleSheet,
    TouchableOpacity,
    ScrollView,
    Image,
    PermissionsAndroid,
    Platform,
    Dimensions,
    FlatList,
} from 'react-native';
import CameraRoll from '@react-native-community/cameraroll';
import {Button, Header} from './commons';
import ImageItems from './ImageItems';
import AntDesign from 'react-native-vector-icons/AntDesign';
// import {useNavigation} from '@react-navigation/native';
import { BGCOLOR, HEADER } from './commons/Colors';
const  hasAndroidPermission = async ()=>{
    const permission = await PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE

    const hasPermission = PermissionsAndroid.check(permission);
    if(await hasPermission){
        return true;
    }
    const status = await PermissionsAndroid.request(permission);
    return status ==='granted';
}

const w = Dimensions.get('window');
const LibraryScreen = ({navigation}) => {
    const [photos, setPhotos] = useState();
    // const navigation = useNavigation();
    const getPhoto = () =>{
        hasAndroidPermission()
        CameraRoll.getPhotos({
            first:20,
            assetType:'Photos',
        })
        .then(r=>{
            setPhotos(r.edges)
        })
        .catch((err)=>{
            console.log('Cant loading photos')
        })
    }
    useEffect(()=>{
        hasAndroidPermission()
        .then(getPhoto)},[])
    return(
        <View style = {styles.container}>
            <Header 
                left = {<AntDesign 
                    onPress = {()=>navigation.goBack()}
                    color={HEADER.text} name = 'arrowleft' size = {22}/>}
                title = {'Library'.toUpperCase()}
                right = {<View/>}
            />
            <FlatList
                data = {photos}
                renderItem = {({item})=>
                    <ImageItems 
                        navigation = {navigation}
                        // source = {{uri:item.node.image.uri}}
                        uri = {item.node.image.uri}
                        fromCamera = {false}
                    />
                    }
                numColumns = {3}
                keyExtractor = {(item, idx)=>idx.toString()}
            />
        </View>
    )
}

export default LibraryScreen;

const styles = StyleSheet.create({
    container:{
        backgroundColor:BGCOLOR,
        flex:1,
    },
    text:{
        color:HEADER.text,
        fontSize:20,
    }
})