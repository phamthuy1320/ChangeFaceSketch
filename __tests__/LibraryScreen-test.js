import React from 'react';
import renderer from 'react-test-renderer';
import LibraryScreen from '../src/screens/LibraryScreen';

test('Library Screen', () => {
  const tree = renderer.create(<LibraryScreen/>).toJSON();
  expect(tree).toMatchSnapshot();
});