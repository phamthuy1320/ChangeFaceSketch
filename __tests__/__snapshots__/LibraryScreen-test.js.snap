// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library Screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f1c3aa",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-between",
          "paddingHorizontal": 5,
        },
        Object {
          "backgroundColor": "#9f4347",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î˜œ
    </Text>
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      LIBRARY
    </Text>
    <View />
  </View>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
